### PR TITLE
[New] Hiveworks

### PR DESCRIPTION
**Pull request by happywillow0**
This is the initial build to attempt #1864 
I can't believe it mostly works. 

Working: 

- Pulls Comic info from HiveWorks catalog
- Since each comic is updated per page, each page is a chapter

Not the best but cannot be fixed:

- The loading screen in between pages / chapters. 

Broken:

- Some web comics do not use the archive page layout that the extension currently built around. It will return 404 or no chapters found. 
- Manga details will not work on backup and restore because there is no details for the webcomic on the webcomic's page. (aka only from the browse screen)

### CODE REVIEW

File: src/en/hiveworks/build.gradle [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    pkgNameSuffix = 'en.hiveworks'
+    extClass = '.HiveWorks'
+    extVersionCode = 1
+    libVersion = '1.2'
+}
+
+dependencies {
+    //compileOnly project(':preference-stub')
+    //compileOnly 'com.github.inorichi.injekt:injekt-core:65b0440'
+}
```
Comment (arkon) [inline src/en/hiveworks/build.gradle]: Remove this

File: src/en/hiveworks/src/eu/kanade/tachiyomi/extension/en/hiveworks/Hiveworks.kt [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+            manga.description = element.select("div.description").text().trim() + "\n" + "\n" + "*Not all comics are supported*"
+            manga.genre = element.select("div.comicrating").text().trim()
+        return manga
+    }
+
+    override fun chapterListParse(response: Response): List<SChapter> {
+        val document = response.asJsoup()
+        val uri = Uri.parse(document.baseUri())
+        val baseUrl = "${uri.scheme}://${uri.authority}"
+        Log.i("TachiDebug", "BaseURL => $baseUrl")
```
Comment (arkon) [inline src/en/hiveworks/src/eu/kanade/tachiyomi/extension/en/hiveworks/Hiveworks.kt]: Remove the logs.

File: src/en/hiveworks/src/eu/kanade/tachiyomi/extension/en/hiveworks/Hiveworks.kt [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+            chapters.add(createChapter(elements[i] , baseUrl))
+        }
+        chapters.reverse()
+        return chapters
+    }
+
+    private fun createChapter(element: Element, baseUrl: String?) = SChapter.create().apply {
+        name = element.text().substringAfter("-").trim()
+        url = "$baseUrl/" + element.attr("value")
+        Log.i("TachiDebug","Chapter URL => $url")
```
Comment (arkon) [inline src/en/hiveworks/src/eu/kanade/tachiyomi/extension/en/hiveworks/Hiveworks.kt]: Same here.

Comment (happywillow0) [reply src/en/hiveworks/src/eu/kanade/tachiyomi/extension/en/hiveworks/Hiveworks.kt]: Sorry, I built these at the same time and I forgot to remove the logs before posting it up. It should be all cleaned up.

**Comment by happywillow0** [comment]
I removed all logs and extra unnecessary comments as far as I'm aware. 

@as280093 , if you get a chance, may I get some icons for this as well?

**Comment by as280093** [comment]
@happywillow0  Here are the icons
[ic_launcher.zip](https://github.com/inorichi/tachiyomi-extensions/files/4007354/ic_launcher.15.zip)

**Pull request merged**